independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/CodestartProcessor.java

```diff
@@ -137,12 +132,25 @@ public void writeFiles() throws IOException {
         for (Map.Entry<String, List<CodestartFile>> e : files.entrySet()) {
             final String relativePath = e.getKey();
             Files.createDirectories(targetDirectory.resolve(relativePath).getParent());
-            getStrategy(relativePath).orElse(CodestartFileStrategyHandler.DEFAULT_STRATEGY)
+            getStrategy(relativePath).orElse(getSelectedDefaultStrategy())
                     .process(targetDirectory, relativePath, e.getValue(), data);
         }
     }
 
-    private Optional<CodestartFileStrategyHandler> getStrategy(final String key) {
+    DefaultCodestartFileStrategyHandler getSelectedDefaultStrategy() {
+        for (CodestartFileStrategy codestartFileStrategy : strategies) {
+            if (Objects.equals(codestartFileStrategy.getFilter(), "*")) {
```

Comment:
What's wrong with `"*".equals(codestartFileStrategy.getFilter())` ? :stuck_out_tongue:

Reply:
I guess the result is the same but I always favor Objects.equals then I don't have to think too much about NPE (even if in that case it's not the case :p)